From e192e4762749cfe3bc350e0353aa82b1a849256d Mon Sep 17 00:00:00 2001
From: bkuhls <bkuhls@users.noreply.github.com>
Date: Sun, 10 Jan 2021 21:56:56 +0100
Subject: [PATCH] meson: optionally depend on c++ (#124)

This fix is needed to allow building with toolchains which lack c++.

Downloaded from upstream commit:
https://github.com/benhoyt/inih/commit/e192e4762749cfe3bc350e0353aa82b1a849256d

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
---
 meson.build | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 4602326..5393af4 100644
--- a/meson.build
+++ b/meson.build
@@ -1,5 +1,5 @@
 project('inih',
-    ['c','cpp'],
+    ['c'],
     default_options : ['default_library=static'],
     license : 'BSD-3-Clause',
 )
@@ -83,6 +83,7 @@ inih_dep = declare_dependency(
 
 #### INIReader ####
 if get_option('with_INIReader')
+    add_languages('cpp')
     inc_INIReader = include_directories('cpp')
 
     lib_INIReader = library('INIReader',
